packages/core/src/Stepper/Step.tsx

```diff
@@ -23,7 +23,7 @@ type BaseProps = HTMLAttributes<BaseElement>
 
 export interface StepContent extends BaseProps {
   /* Step's label */
-  readonly label: string
+  readonly label: string | ReactNode
```

Comment:
Do not use both, `string` and `ReactNode` as this may lead to bugs in future. Either use only `ReactNode` or only `string`. You would also want to export `StepLabel ` when using `ReactNode`.

Reply:
Got no response so not sure if PR is updated but [documentation](https://github.com/AxisCommunications/practical-react-components/blob/ffd63731a4f4c7d9649eca36877e05272344124a/packages/docs/src/mdx/coreComponents/Stepper.mdx) and [tests](https://github.com/AxisCommunications/practical-react-components/blob/a420596cae9aa411db7c641f76d8a69ed1ce9a53/packages/core/src/Stepper/index.test.tsx) should be updated as well and a proper component should be used instead of the string.

Reply:
You should use `StepLabel` instead of `p` for the `Step`'s label as that was the purpose of exporting the component. You also need to change [this code](https://github.com/AxisCommunications/practical-react-components/blob/33daa7d131a995708186534bd6e589e7dc994e99/packages/core/src/Stepper/Step.tsx#L280) so that the `ReactNode` is not wrapped in `StepLabel`  by default as you may get `div` in `p` otherwise.

Reply:
IMO you do not need a complex solution. I was thinking about something like below (diff):

```
diff --git a/packages/core/src/Stepper/Step.tsx b/packages/core/src/Stepper/Step.tsx
index 65676b3..42dec0b 100644
--- a/packages/core/src/Stepper/Step.tsx
+++ b/packages/core/src/Stepper/Step.tsx
@@ -23,7 +23,7 @@ type BaseProps = HTMLAttributes<BaseElement>

 export interface StepContent extends BaseProps {
   /* Step's label */
-  readonly label: string
+  readonly label: ReactNode
   /* Step's content */
   readonly content: ReactNode
   /* Indicates whether the step has errors that should be resolved */
@@ -110,7 +110,9 @@ const StepHeader = styled.div`
   margin-top: -${spacing.large};
 `

-const StepLabel = styled(Typography)<{
+export const StepLabel = styled(Typography).attrs({ variant: 'default-text' })``
+
+const StepLabelWrapper = styled.div<{
   readonly active: boolean
   readonly completed: boolean
 }>`
@@ -277,9 +279,9 @@ export const Step: FC<StepProps> = ({
         {lastStep ? null : (
           <StepDivider active={active} completed={completed} />
         )}
-        <StepLabel variant="default-text" active={active} completed={completed}>
+        <StepLabelWrapper active={active} completed={completed}>
           {label}
-        </StepLabel>
+        </StepLabelWrapper>
         {active ? (
           <StepContentContainer>{children}</StepContentContainer>
         ) : null}
diff --git a/packages/core/src/Stepper/index.test.tsx b/packages/core/src/Stepper/index
.test.tsx
index e3bced5..47c6625 100644
--- a/packages/core/src/Stepper/index.test.tsx
+++ b/packages/core/src/Stepper/index.test.tsx
@@ -2,7 +2,7 @@ import 'jest-styled-components'
 import { expect, test, describe } from '@jest/globals'

 import { TestRender } from '../TestUtils'
-import { Stepper } from '.'
+import { Stepper, StepLabel } from '.'

 const NOOP = () => {
   /** */
@@ -14,16 +14,16 @@ describe('Stepper', () => {
       <Stepper
         steps={[
           {
-            label: 'Step 1 Title',
+            label: <StepLabel>Step 1 Title</StepLabel>,
             content: <div />,
           },
           {
-            label: 'Step 2 Title',
+            label: <StepLabel>Step 2 Title</StepLabel>,
             content: <div />,
             hasErrors: true,
           },
           {
-            label: 'Step 3 Title (done)',
+            label: <StepLabel>Step 3 Title (done)</StepLabel>,
             content: <div />,
           },
         ]}
@@ -51,11 +51,11 @@ describe('Stepper', () => {
       <Stepper
         steps={[
           {
-            label: 'Step 1 Title',
+            label: <StepLabel>Step 1 Title</StepLabel>,
             content: <div />,
           },
           {
-            label: 'Step 3 Title (done)',
+            label: <StepLabel>Step 3 Title (done)</StepLabel>,
             content: <div />,
           },
         ]}
@@ -83,7 +83,7 @@ describe('Stepper', () => {
       <Stepper
         steps={[
           {
-            label: 'Step 1 Title',
+            label: <StepLabel>Step 1 Title</StepLabel>,
             content: <div />,
           },
         ]}
diff --git a/packages/core/src/Stepper/index.tsx b/packages/core/src/Stepper/index.tsx
index 678eaef..448da40 100644
--- a/packages/core/src/Stepper/index.tsx
+++ b/packages/core/src/Stepper/index.tsx
@@ -18,7 +18,7 @@ const StepperWrapper = styled(Paper)`
   box-shadow: none;
 `

-export { StepContent, StepperAction } from './Step'
+export { StepContent, StepperAction, StepLabel } from './Step'

 type BaseElement = HTMLDivElement
 type BaseProps = HTMLAttributes<BaseElement>
diff --git a/packages/docs/src/mdx/coreComponents/Stepper.mdx b/packages/docs/src/mdx/c
oreComponents/Stepper.mdx
index b74911c..d69a8ce 100644
--- a/packages/docs/src/mdx/coreComponents/Stepper.mdx
+++ b/packages/docs/src/mdx/coreComponents/Stepper.mdx
@@ -10,6 +10,7 @@ import {
   Typography,
   RadioButtonGroupField,
   Stepper,
+  StepLabel,
 } from 'practical-react-components-core'

 # Stepper
@@ -35,7 +36,7 @@ export const DemoComponent = ({}) => {
       className="my-stepper"
       steps={[
         {
-          label: 'Step 1 Title',
+          label: <StepLabel>Step 1 Title</StepLabel>,
           content: (
             <div>
               <RadioButtonGroupField
@@ -50,7 +51,7 @@ export const DemoComponent = ({}) => {
           className: 'my-step',
         },
         {
-          label: 'Step 2 Title',
+          label: <StepLabel>Step 2 Title</StepLabel>,
           content: (
             <div>
               <TextInputField label="Name" value="" placeholder="First name" />
@@ -58,7 +59,7 @@ export const DemoComponent = ({}) => {
           ),
         },
         {
-          label: 'Step 3 Title (done)',
+          label: <StepLabel>Step 3 Title (done)</StepLabel>,
           content: (
             <Typography variant="navigation-label">You are now done</Typography>
           ),
@@ -92,7 +93,7 @@ export const DemoComponent = ({}) => {
 <Stepper
   steps={[
     {
-      label: 'Step 1 Title',
+      label: <StepLabel>Step 1 Title</StepLabel>,
       content: (
         <div>
           <RadioButtonGroupField
@@ -109,7 +110,7 @@ export const DemoComponent = ({}) => {
       ),
     },
     {
-      label: 'Step 2 Title',
+      label: <StepLabel>Step 2 Title</StepLabel>,
       content: (
         <div>
           <TextInputField label="Name" value="" placeholder="First name" />
@@ -117,7 +118,7 @@ export const DemoComponent = ({}) => {
       ),
     },
     {
-      label: 'Step 3 Title (done)',
+      label: <StepLabel>Step 3 Title (done)</StepLabel>,
       content: (
         <Typography variant="navigation-label">You are now done</Typography>
       ),
```

Reply:
done

Reply:
done

Reply:
LGTM now